testkit/src/main/scala/org/bitcoins/testkit/core/gen/CreditingTxGen.scala

```diff
@@ -350,6 +351,20 @@ sealed abstract class CreditingTxGen {
         }
       }
     }
+
+  def inputsAndOuptuts: Gen[
+    (Seq[BitcoinUTXOSpendingInfo], Seq[TransactionOutput])] =
+    outputs
+      .flatMap { creditingTxsInfo =>
+        val creditingOutputs = creditingTxsInfo.map(c => c.output)
+        val creditingOutputsAmt = creditingOutputs.map(_.value)
+        val totalAmount = creditingOutputsAmt.fold(CurrencyUnits.zero)(_ + _)
+
+        TransactionGenerators.smallOutputs(totalAmount).map { destinations =>
+          (creditingTxsInfo, destinations)
+        }
+      }
+      .suchThat(_._1.nonEmpty)
```

Comment:
nit it's nice to avoid the use of `.suchThat()` in simple cases like this where you can make another generator pretty easily.

Reply:
So actually in this case it should always be the case that the list is non-empty since it uses a number between 1 and 5 ... but then if things fail and it tries to shrink for some reason it occasionally uses an empty list and gives that error message instead, and this seems to fix that.